### ISSUE TITLE
XForms Performance issue

### ISSUE THREAD

**Issue by setuncay** [orbeon/orbeon-forms#5871]
I'm running some load tests on 2022.1.3.202304130216 PE under tomcat. My tomcat has 32 GB of RAM 16 Cpus. Xmx and Xms settings of the Tomcat are also adjusted accordingly. My test form is pretty basic with approximately 8 fields (name, surname, etc). 

My Apache JMeter script basically loads the form, sets the focus on the name field, updates the value of the name field and performs these steps for the other fields in the form and eventually submits it. 

When I run the test with up to 18 users (ramp up duration of 10 secs), everything works fine and Orbeon responds without any issues. Starting with 18 users, Orbeon is acting strangely and doesn't process  requests even though there is plenty of free resources on the server / tomcat.

When I send a request to Orbeon in this 'stuck' state, I can see a log line similar to:

`2023-06-18T14:55:01,672 INFO  lifecycle - event: {"request": "1422", "session": "44C591AB724587B7BCEC88788C32BB57", "source": "limiter", "message": "start: filter", "path": "/fr/xxx/loadtest/new", "method": "GET"}`

and that line is the last line for the corresponding request id and I don't see any other 'lifecycle' log lines for it.

As mentioned CPU / Memory / Tomcat Thread Pool settings etc.. are all fine at that point and there are plenty of resources for orbeon.

Any suggestions to figure out the underlying reason is really appreciated.

**Comment by avernet** [comment]
You are hitting the Limiter Filter, which is trying to limit the number of concurrent requests being processed at the same time based on the number of cores available. You can read more about what it tries to do in the [doc](https://doc.orbeon.com/configuration/advanced/limiter-filter). Depending on your setup and what your forms are doing, you're better off disabling it, so I'd recommend you try to comment out the corresponding `<filter-mapping>` in the `web.xml`. You'll let me know if this helps.

**Comment by setuncay** [comment]
Thanks very much @avernet . I will re-try after disabling the Limiter filter. Another question though - Looks like once the Limiter kicks in, it keeps blocking requests even when there is no active sessions left. Is this by design?

**Comment by avernet** [comment]
@setuncay When the max number of concurrent threads is reached, any additional request will have to wait for one of the requests that is currently using a thread to be done. One way to put it is that any new request beyond `max-threads` is "queued".

A typical case where this heuristic results in a suboptimal use of resources is when to process a request you need to do something that takes a significant amount of time, but isn't CPU-bound, like calling a service or running a SQL query on a database server.

And you'll let us know if you're getting better results without the Limiter Filter.

**Comment by setuncay** [comment]
Thanks again @avernet . What is puzzling is that processing threads never seem to be unblocked. My session timeout is set to 60 minutes in Tomcat, and here is what happened when the Limit Filter with its default settings are active:

1. 18 users create concurrently create a new form
2. 18 users concurrently updates field values
3. Delimiter filter kicks in blocking the forms processing
4. All requests are killed in JMeter

After step 4, I'd expect orbeon starts accepting requests (at least after 60 mins when all existing sessions are expired). However, that is not the case and I have to bounce the tomcat to process new requests.

Can you please shed some light on this behaviour?

Thanks!

**Comment by avernet** [comment]
@setuncay Yes, this is most surprising. Would you be able to reproduce this issue on a form you create on https://demo.orbeon.com/, and then link to your form here, and attach your JMeter file, so we can reproduce the issue?

**Comment by setuncay** [comment]
@avernet the issue of stuck threads once we start to use "org.apache.tomcat.jdbc.pool.DataSourceFactory" DB connection pool rather than the default bcp2 pool. Now we hit another issue, our test system consists multiple tomcat wrapped orbeon instances behind an AWS ALB with sticky session configured. Sometimes, some of our requests are failed with 403 and when we check the orbeon log for them we see the following:

```
2023-06-22T08:53:16,390 WARN  ResponseProcessCookies - Invalid cookie header: "Set-Cookie: AWSALBAPP-3=_remove_; Expires=Thu, 29 Jun 2023 08:53:16 GMT; Path=/". Invalid 'expires' attribute: Thu, 29 Jun
2023 08:53:16 GMT
```

For some reason Orbeon fails to parse the 'expires' attribute resulting in 403 response. Can you please see why this is the case?

Thanks!

**Comment by avernet** [comment]
@setuncay We use GitHub Issue to keep track of verified bugs or request for enhancement, not for support. So I've started a new thread about this on your Basecamp, and suggest we continue the discussion [there](https://3.basecamp.com/3600924/buckets/7993860/messages/6283201720).
